lib/engine/generator.js

```diff
@@ -41,8 +43,18 @@ exports.getCombinations = function *(components, metadata, routes, path, options
        * the component we're dealing with, which could massively
        * reduce the cartesian product we need to do afterwards.
        */
-      const matchingRules = rules.filter((rule) => {
-        return rule.fn && _.isEqual(rule.scope, [ component.name ])
+
+      const rulesBypage = rules.filter((rule) => {
+        if (isEmpty(rule.pages)) {
+          return true
+        }
+
+        // At least one path from the rules, should match the active one
+        return rule.pages.some(page => page === path.join('/'))
```

Comment:
Can we make this:

```js
return rule.pages.some(page => `/${page}` === path.join('/'))
```

so that the rules are consistent in requiring a `/` prefix?

For example: `pages: [ '/', '/contributors' ]` rather than [ '/', 'contributors' ]`